File: Level 2/서버 증설 횟수/solution.py
def solution(players, m, k):
    servers = [0] * 24

    for i in range(len(players)):
        servers_range = servers[i - k + 1 : i] if i - k + 1 >= 0 else servers[:i]
        added_servers = sum(servers_range)
        servers_required = players[i] // m

        if added_servers < servers_required:
            servers[i] = servers_required - added_servers

    return sum(servers)

File: Level 2/서버 증설 횟수/solution.js
function solution(players, m, k) {
  const servers = new Array(24).fill(0);

  players.forEach((player, i) => {
    const serversRange =
      i - k + 1 >= 0 ? servers.slice(i - k + 1, i) : servers.slice(0, i);
    const serversRequired = Math.floor(player / m);
    const addedServers = sum(serversRange);

    if (addedServers < serversRequired) {
      servers[i] = serversRequired - addedServers;
    }
  });

  return sum(servers);
}

function sum(arr) {
  return arr.reduce((acc, cur) => acc + cur, 0);
}
